cmd/dep/govend_importer.go

```diff
@@ -0,0 +1,169 @@
+	"github.com/golang/dep/internal/gps"
+	"github.com/pkg/errors"
+)
+
+// ToDo: govend supports json and xml formats as well and we will add support for other formats in next PR - @RaviTezu
+// govend don't have a separate lock file.
+const govendYAMLName = "vendor.yml"
+
+// govendImporter imports govend configuration in to the dep configuration format.
+type govendImporter struct {
+	yaml govendYAML
+
+	logger  *log.Logger
+	verbose bool
+	sm      gps.SourceManager
+}
+
+func newGovendImporter(logger *log.Logger, verbose bool, sm gps.SourceManager) *govendImporter {
+	return &govendImporter{
+		logger:  logger,
+		verbose: verbose,
+		sm:      sm,
+	}
+}
+
+type govendYAML struct {
+	Imports []govendPackage `yaml:"vendors"`
+}
+
+type govendPackage struct {
+	Path     string `yaml:"path"`
+	Revision string `yaml:"rev"`
+}
+
+func (g *govendImporter) Name() string {
+	return "govend"
+}
+
+func (g *govendImporter) HasDepMetadata(dir string) bool {
+	y := filepath.Join(dir, govendYAMLName)
+	if _, err := os.Stat(y); err != nil {
+		return false
+	}
+
+	return true
+}
+
+func (g *govendImporter) Import(dir string, pr gps.ProjectRoot) (*dep.Manifest, *dep.Lock, error) {
+	err := g.load(dir)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return g.convert(pr)
+}
+
+// load the govend configuration files.
+func (g *govendImporter) load(projectDir string) error {
+	g.logger.Println("Detected govend configuration files...")
+	y := filepath.Join(projectDir, govendYAMLName)
+	if g.verbose {
+		g.logger.Printf("	Loading %s", y)
+	}
+	yb, err := ioutil.ReadFile(y)
+	if err != nil {
+		return errors.Wrapf(err, "Unable to read %s", y)
+	}
+	err = yaml.Unmarshal(yb, &g.yaml)
+	if err != nil {
+		return errors.Wrapf(err, "Unable to parse %s", y)
+	}
+	return nil
+}
+
+// convert the govend configuration files into dep configuration files.
+func (g *govendImporter) convert(pr gps.ProjectRoot) (*dep.Manifest, *dep.Lock, error) {
+	g.logger.Println("Converting from vendor.yaml...")
+
+	manifest := dep.NewManifest()
+	lock := &dep.Lock{}
+
+	for _, pkg := range g.yaml.Imports {
+		// Path must not be empty
+		if pkg.Path == "" || pkg.Revision == "" {
+			return nil, nil, errors.New("Invalid govend configuration, Path or Rev is required")
+		}
+
+		p, err := g.sm.DeduceProjectRoot(pkg.Path)
+		if err != nil {
+			return nil, nil, err
+		}
+		pkg.Path = string(p)
+
+		// Check if the current project is already existing in locked projects.
+		if projectExistsInLock(lock, p) {
+			continue
+		}
+
+		pi := gps.ProjectIdentifier{
+			ProjectRoot: gps.ProjectRoot(pkg.Path),
+		}
+		revision := gps.Revision(pkg.Revision)
+
+		version, err := lookupVersionForLockedProject(pi, nil, revision, g.sm)
+		if err != nil {
+			g.logger.Println(err.Error())
+		} else {
+			pp := getProjectPropertiesFromVersion(version)
+			if pp.Constraint != nil {
+				pc, err := g.buildProjectConstraint(pkg, pp.Constraint.String())
+				if err != nil {
```

Comment:
I realize this isn't in the other importers yet, but we'd like to have import keep going even when we can't guess a constraint from the revision. See #907.

So when `buildProjectConstraint` returns an error, let's log a warning like "Unable to infer a constraint for REVISION", and then skip adding the constraint to the manifest.